File: easy/Age-Distribution/Age Distribution.cs
using System;
using System.IO;
using System.Linq;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            ShowPhrase(line);
        }
    }
    static void ShowPhrase(string line){
        int age = Convert.ToInt32(line);
        var options = new Dictionary<Func<int,bool>,Action>{
            {ag => 0<=ag && ag<=2, () => Console.WriteLine("Still in Mama's arms")},
            {ag => 3<=ag && ag<=4, () => Console.WriteLine("Preschool Maniac")},
            {ag => 5<=ag && ag<=11,() => Console.WriteLine("Elementary school")},
            {ag => 12<=ag && ag<=14,() => Console.WriteLine("Middle school")},
            {ag => 15<=ag && ag<=18,() => Console.WriteLine("High school")},
            {ag => 19<=ag && ag<=22,() => Console.WriteLine("College")},
            {ag => 23<=ag && ag<=64,() => Console.WriteLine("Working for the man")},
            {ag => 66<=ag && ag<=100,() => Console.WriteLine("The Golden Years")},
            {ag => 0>ag||ag>66,() =>Console.WriteLine("This program is for humans")},
        };
        options.First(kvp => kvp.Key(age)).Value();
    }
}

File: easy/Age-Distribution/Age Distribution.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            System.out.println(ShowPhrase(line));
        }
    }
    
    static String ShowPhrase(String line){
        int age = Integer.valueOf(line);
        if(age<0 || age>100) return "This program is for humans";
        if(age<3) return "Still in Mama's arms";
        if(age<5) return "Preschool Maniac";
        if(age<12) return "Elementary school";
        if(age<15) return "Middle school";
        if(age<19) return "High school";
        if(age<23) return "College";
        if(age<66) return "Working for the man";
        return "The Golden Years";
    }
}
